sys/auto_init/saul/auto_init_bme280.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2016 Kees Bakker, SODAQ
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     auto_init_saul
+ * @{
+ *
+ * @file
+ * @brief       Auto initialization of BME280 driver.
+ *
+ * @author      Kees Bakker <kees@sodaq.com>
+ *
+ * @}
+ */
+
+#ifdef MODULE_BME280
+
+#include "log.h"
+#include "saul_reg.h"
+
+#include "bme280_params.h"
+#include "bme280.h"
+
+/**
+ * @brief   Define the number of configured sensors
+ */
+#define BME280_NUMOF    (sizeof(bme280_params) / sizeof(bme280_params[0]))
+
+/**
+ * @brief   Allocation of memory for device descriptors
+ */
+static bme280_t bme280_devs[BME280_NUMOF];
+
+/**
+ * @brief   Memory for the SAUL registry entries
+ */
+static saul_reg_t saul_entries[BME280_NUMOF * 3];
+
+void auto_init_bme280(void)
+{
+    size_t se_ix = 0;
+    for (size_t i = 0; i < BME280_NUMOF; i++) {
+        int res = bme280_init(&bme280_devs[i], &bme280_params[i]);
+        if (res < 0) {
+            LOG_ERROR("Unable to initialize BME280 sensor #%i\n", i);
+        }
+        else {
+            /* temperature */
+            saul_entries[se_ix].dev = &bme280_devs[i];
+            saul_entries[se_ix].name = "bme280-temp";
```

Comment:
See the changes in #6385, please add a `saul_reg_info_t` table in `drivers/bme280/include/bme280_params.h`.